app/controllers/comment_controller.php

```diff
@@ -0,0 +1,64 @@
+<?php
+class CommentController extends AppController
+{
+    function __construct($name)
+    {
+        parent::__construct($name);
+
+        if (!is_logged_in()) 
+        {
+            redirectTo(url('user/login'));
+        }
+    }
+
+    //Display thread and it's comments
+    public function view()
+    {
+        $thread     = Thread::get(Param::get('thread_id'));
+        $cur_page   = max(Param::get('page'), SimplePagination::MIN_PAGE_NUM);
+        $pagination = new SimplePagination($cur_page, MAX_ITEM_DISPLAY);
+        
+        $comments        = Comment::getAll($thread->id);
+        
+        $other_commments = array_slice($comments, $pagination->start_index + SimplePagination::MIN_PAGE_NUM);
+        
+        $pagination->checkLastPage($other_commments);
+
+        $page_links = createPageLinks(count($comments), $cur_page, $pagination->count, 'thread_id=' . $thread->id);
+
+        $comments   = array_slice($comments, $pagination->start_index -1, $pagination->count);
+
+        $this->set(get_defined_vars());
+    }
+
+    //Add new comment
+    public function write()
+    {    
+        $thread     = Thread::get(Param::get('thread_id'));
+        $comment    = new Comment;
+        $page       = Param::get('page_next', 'write');
+
+        switch ($page){
```

Comment:
Added po.

*Update*
A more recent comment contradicted this one, I stuck with the more recent one. Thank you.